services/src/api/handlers/datasets.rs

```diff
@@ -354,6 +359,57 @@ pub async fn update_dataset_symbology_handler<C: ApplicationContext>(
     Ok(HttpResponse::Ok())
 }
 
+// Updates the dataset's provenance
+#[utoipa::path(
+    tag = "Datasets",
+    put,
+    path = "/dataset/{dataset}/provenance",
+    request_body = Provenances,
+    responses(
+        (status = 200, description = "OK"),
+        (status = 400, description = "Bad request", body = ErrorResponse),
+        (status = 401, response = crate::api::model::responses::UnauthorizedUserResponse)
+    ),
+    params(
+        ("dataset" = DatasetName, description = "Dataset Name"),
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+pub async fn update_dataset_provenance_handler<C: ApplicationContext>(
+    session: C::Session,
+    app_ctx: web::Data<C>,
+    dataset: web::Path<DatasetName>,
+    provenance: ValidatedJson<Provenances>,
+) -> Result<impl Responder> {
```

Comment:
what is the correct response type then in this case?

Reply:
it does return something so () does not work
it does not return a tuple so ((), StatusCode) does not work
I don't get how HttResponse is better. Why is a Response better than a Responder?

Reply:
When you change the return type, you will know.
If any responder can be returned, the API breaks more easily.

Reply:
I don't get it. A `Responder` is something that can be turned into a HTTP Response, which is exactly what actix does with the result of the handler function.

But here you 5a4c0eac6961fcf8445ce061a001ed120fcfbd1c

Previously you could return something that could be turned into a Response, now I return the Response directly. The response, which can have any header and any payload :shrug:

Reply:
But this is up to you and not my suggestion that you've selected the most lamest refactoring.

For instance, you could have returned `Ok(())` and specified this as `Result<()>`. This would be 200 OK without a body.

You can decide on this.

Reply:
> For instance, you could have returned Ok(()) and specified this as Result<()>. This would be 200 OK without a body.

I get an error:

> the trait bound `(): actix_web::Responder` is not satisfied